### DISCUSSION TITLE
ARC - Manual Workflow Trigger

### DISCUSSION THREAD

**Discussion by devops-sailboat** [actions/actions-runner-controller#2575] [Questions]
Hi everyone,

I would like to know how we can incorporate a manual workflow trigger for the ARC. The goal is to ensure that when a CI/CD process is initiated, it will wait for an approval or trigger before proceeding with the workflow execution.

This manual trigger will allow us to have more control over the release process and ensure that the necessary approvals or triggers are in place before each workflow run.

If you have any insights or suggestions on how to implement this manual workflow trigger for the ARC, please share your thoughts.

example:

name: My First Manual Workflow
on: [workflow_dispatch]
jobs:
  example:
    name: Example Job
    runs-on: ubuntu-latest

<img width="856" alt="Screenshot 2023-05-10 at 5 00 01 PM" src="https://github.com/actions/actions-runner-controller/assets/95968851/4bcee836-e9e8-40ed-85df-482f32952faf">

**Comment by nobbs** [comment]
If I understood you correctly, you want to add an "approval step" before running the workflow, right? If so, then take a look at [environments](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment), they provide the feature you're looking for.

**Reply by AzarOPS** [reply]
Environments is supported only for Public repositories. Or Enterprise plan. Any other suggestion for Private repositories ?
